explorer/ast/value.h

```diff
@@ -1593,11 +1594,12 @@ class StaticArrayType : public Value {
   }
 
   auto element_type() const -> const Value& { return *element_type_; }
-  auto size() const -> size_t { return size_; }
+  auto size() const noexcept -> std::optional<size_t> { return size_; }
```

Comment:
```suggestion
  auto size() const ->size_t { CARBON_CHECK(has_size()); return *size_; }
```

Mostly I'm commenting here because `size()` is returning an optional, and there's also `has_size()`, so that's really providing two ways of checking existence. I'd be fine with removing `has_size`, but I'm suggesting an edit in this direction since you currently use `has_size` and this seems more consistent with `has_size_expression` in `ArrayTypeLiteral` (which is, I think, the trend of how explorer code has gone).